packages/alchemy/src/react/hooks/useAuthenticate.ts

```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import { useMutation } from "@tanstack/react-query";
+import { ClientOnlyPropertyError } from "../../config/errors.js";
+import type { AuthParams } from "../../signer/signer.js";
+import { useAlchemyAccountContext } from "../context.js";
+import { useSigner } from "./useSigner.js";
+
+export function useAuthenticate() {
+  const { queryClient } = useAlchemyAccountContext();
+  const signer = useSigner();
+
+  const {
+    mutate: authenticate,
+    isPending,
+    error,
+  } = useMutation(
+    {
+      mutationFn: async (authParams: AuthParams) => {
+        if (!signer) {
+          throw new ClientOnlyPropertyError("signer");
+        }
+
+        const user = await signer.authenticate(authParams);
+
+        return user;
```

Comment:
if you want you can just return the promise right?
```suggestion
        return signer.authenticate(authParams);
```

Reply:
oh yea definitely... not sure why I was doing this